src/components/Services.jsx

```diff
@@ -57,6 +57,24 @@ export const Services = ({ installedKonnectors, suggestedKonnectorsQuery }) => {
     hasZeroInstalledKonnectors &&
     (suggestedKonnectors.length >= 1 ||
       fallbackKonnectorSuggestions.length >= 1)
+  const { data: jobData } = useQuery(
+    Q('io.cozy.jobs')
+      .where({ worker: 'konnector' })
+      .sortBy([{ started_at: 'desc' }]),
```

Comment:
Since the `where` condition is "static", it's better to use a partial index for this, especially since the jobs database can be very large.
Also, it's better to explicitly index fields to prevent unexpected behaviour coming from the automatic cozy-client indexing.
And since you filter the running jobs afterwards, you might add `state: 'running'` in the partial index? That would considerably speed up your query and decrease the number of docs returned.

```js
  const { data: jobData } = useQuery(
    Q('io.cozy.jobs')
      .indexFields(['started_at'])
      .sortBy([{ started_at: 'desc' }])
      .partialIndex({ worker: 'konnector', `state: 'running'` })
```

Reply:
Indeed thanks a lot, I'm quite inexperienced with queries unfortunately and was seeking that type of feedback from you! Updating it

Reply:
@paultranvan I implemented the modifications but I get this persistent error (though it doesn't impact me when a connector starts running, it's only at app launch):

![image](https://user-images.githubusercontent.com/12577784/224953575-fea02dde-df05-4544-8a52-32eec04077a0.png)

Reply:
You have to upgrade cozy-client https://github.com/cozy/cozy-client/commit/2e770e35c0037def9a48b3170c5419e3f5ebe686

Reply:
@Crash-- but I did, didn't I? https://github.com/cozy/cozy-home/pull/1899/commits/407a63441307703ae3fe70dd9750ec2db0e30124

Reply:
oh I have the same bug from yesterday
```
  "name": "cozy-client",
  "version": "35.3.1",
  "license": "MIT",
  "main": "dist/index.node.js",
  "browser": "dist/index.js",
  "types": "types/index.d.ts",
```
the old version is in node modules but the new version is in yarn.lock and package.json... I have to find what's the source of this issue (it's definitely linked to my env)

Reply:
So this issue is solved and the request is now optimized, right?

Reply:
yes it should be given the current parameters. I'm still mapping jobs to string[], I'm not sure if the query can do that @paultranvan

Reply:
I'm not sure why you need a mapping to string[], but the query simply gives you an object containing a data array with the docs inside, without typing: https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/DocumentCollection.js#L316-L321

Reply:
@acezard You can use .select() to only select the information you want. It seems that you only want `job.message.konnector` so you can do something like: 

```
Q('io.cozy.jobs')
.select(['message.konnector'])
.indexFields(['started_at'])
    .sortBy([{ started_at: 'desc' }])
    .partialIndex({ worker: 'konnector', state: 'running' }),
  options: { as: 'io.cozy.jobs/running/konnector/messageKonnector' }
```

And you should only get data = string[]

Reply:
I can't seem to make this work, it doesn't return Array<KonnectorSlug>

Reply:
> I can't seem to make this work

What does that mean? Is the request OK or KO? Does it add a few fields that you don't request? Like `_id` or something ?

Reply:
it complains that select is missing fields used in indedFields then when added it returns an array of objects containing the jobs

Reply:
Seen with @acezard: this response is actually not the mango query itself but coming from the realtime, that fills the store with the whole doc. The query looks ok now